client/src/hooks/mutations/useCalculateRewards.ts

```diff
@@ -0,0 +1,30 @@
+import { useMutation, UseMutationOptions, UseMutationResult } from '@tanstack/react-query';
+import { useCallback, useRef } from 'react';
+
+import { PostCalculateRewardsResponse, apiPostCalculateRewards } from 'api/calls/calculateRewards';
+
+type CalculateRewardsArguments = {
+  amountGlm: string;
+  numberOfDays: number;
+};
+
+export default function useCalculateRewards(
+  options?: UseMutationOptions<any, unknown, CalculateRewardsArguments>,
+): UseMutationResult<PostCalculateRewardsResponse, unknown, CalculateRewardsArguments> {
+  const abortControllerRef = useRef<AbortController | null>(null);
+
+  const mutation = useMutation({
+    mutationFn: async ({ amountGlm, numberOfDays }) => {
+      abortControllerRef.current = new AbortController();
+      return apiPostCalculateRewards(amountGlm, numberOfDays, abortControllerRef.current.signal);
+    },
+    ...options,
+  });
+
+  const reset = useCallback(() => {
+    abortControllerRef.current?.abort();
+    mutation.reset();
+  }, [abortControllerRef, mutation]);
```

Comment:
deps array can be empty in this case.

Reply:
Any harm in having that? It won't redeclare and linter is happy.

Reply:
reference doesn't call rendering, so it is unnecessary, mutation is an object, which means it calls a new reset method every time.

Reply:
Done. Removed from `useAllocateSimulate` too.